### PR TITLE
Adding scanpy recipe

### PR DESCRIPTION
**Pull request by flying-sheep**
CC @conda-forge/help-python @ivirshup

<details>
<summary>Checklist</summary>

- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

</details>

Bioconda /discussions link: https://github.com/orgs/conda-forge/teams/bioconda-recipes/discussions/13

### CODE REVIEW

**Comment by flying-sheep** [comment]
How is this a conflict?

```py
Package h5py conflicts for:
anndata[version='>=0.7.4'] -> h5py
h5py[version='>=2.10']
```

File: recipes/scanpy/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    - patsy
+    - networkx >=2.3
+    - natsort
+    - joblib
+    - numba >=0.41
+    - umap-learn >=0.3.10
+    - legacy-api-wrap
+    - packaging
+    - sinfo
+    - importlib_metadata  # [py<38]
```
Comment (chrisburr) [inline recipes/scanpy/meta.yaml:46]: See https://conda-forge.org/docs/maintainer/knowledge_base.html#empty-python-packages
```suggestion
    - importlib_metadata
```

File: recipes/scanpy/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+package:
+  name: {{ name }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: ad5e7afdf1948753cd27f6f27ee904df1ce455855eec8520985effe86c53c630
+
+build:
+  # noarch: python  # for some reason the [py<38] makes it not work?
```
Comment (chrisburr) [inline recipes/scanpy/meta.yaml:13]:
```suggestion
  noarch: python
```

**Comment by chrisburr** [comment]
For the conflict:

```
$ conda mambabuild recipes/scanpy --python 3.9
...
Mamba failed to solve:
 - pip
 - legacy-api-wrap
 - h5py >=2.10
 - networkx >=2.3
 - numpy >=1.17
 - umap-learn >=0.3.10
 - numba >=0.41
 - pandas >=0.21
 - packaging
 - tqdm
 - sinfo
 - natsort
 - joblib
 - pytables
 - patsy
 - seaborn
 - matplotlib >=3.1.2
 - anndata >=0.7.4
 - scipy >=1.4
 - statsmodels >=0.10
 - python >=3.9,<3.10.0a0
 - scikit-learn >=0.21.2

with channels:

The reported errors are:
   Encountered problems while solving:
     - package anndata-0.7.4-py37hc8dfbb8_0 requires python >=3.7,<3.8.0a0, but none of the providers can be installed
```

In this case the error is misleading (errors from SAT solvers are hard) but it can be improved by iteratively removing dependencies until it's a minimal reproducer:
```bash
$ mamba create --name test 'pip' 'legacy-api-wrap' 'h5py >=2.10' 'networkx >=2.3' 'numpy >=1.17' 'umap-learn >=0.3.10' 'numba >=0.41' 'pandas >=0.21' 'packaging' 'tqdm' 'sinfo' 'natsort' 'joblib' 'pytables' 'patsy' 'seaborn' 'matplotlib >=3.1.2' 'anndata >=0.7.4' 'scipy >=1.4' 'statsmodels >=0.10' 'python >=3.9,<3.10.0a0' 'scikit-learn >=0.21.2'
Encountered problems while solving:
  - package anndata-0.7.4-py37hc8dfbb8_0 requires python >=3.7,<3.8.0a0, but none of the providers can be installed
$ mamba create --name test 'umap-learn >=0.3.10' 'python =3.9'
Encountered problems while solving:
  - package umap-learn-0.3.10-py27_0 requires python >=2.7,<2.8.0a0, but none of the providers can be installed
```

The issue is that `umap-learn` hasn't been built for Python 3.9. As you can make this `noarch: python` you don't need to worry about it but it should be fixed by this PR: https://github.com/conda-forge/umap-learn-feedstock/pull/31

File: recipes/scanpy/meta.yaml (hunk 1 above)
Comment (flying-sheep) [reply recipes/scanpy/meta.yaml:46]: I see

> Currently available packages:
> 
> Name | Available on: | Empty on:
> -- | -- | --
> dataclasses | python &gt;=3.6,&lt;3.7 | python &gt;=3.7
> typing | &nbsp; | python &gt;=3

no mention of `importlib_metadata`.

File: recipes/scanpy/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  run:
+    - python
+    - anndata >=0.7.4
+    - numpy >=1.17
+    - matplotlib >=3.1.2
+    - pandas >=0.21
+    - scipy >=1.4
+    - seaborn
+    - h5py >=2.10
+    - pytables  # “tables” on PyPI
```
Comment (bdice) [inline recipes/scanpy/meta.yaml:33]:
```suggestion
    - pytables  # "tables" on PyPI
```

**Review by bdice — COMMENTED** [review]
Thanks @flying-sheep for working on this. I'm also interested in installing scanpy via conda-forge.

File: recipes/scanpy/meta.yaml (hunk 1 above)
Comment (flying-sheep) [reply recipes/scanpy/meta.yaml:46]: @chrisburr ping

**Comment by ivirshup** [comment]
Is there any kind of error message for the windows build? I can't see anything particularly useful in the log. If not, could be worth checking if it's 3.6 specific, then dropping that?

**Comment by flying-sheep** [comment]
I also don’t know what that failure is … :shrug:

**Comment by flying-sheep** [comment]
Ha, works!

File: recipes/scanpy/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    - python >=3.6
+    - pip
+    - flit-core
+    - setuptools_scm
+    - pytoml
+  run:
+    - python >=3.6
+    - anndata >=0.7.4
+    - numpy >=1.17
+    - matplotlib >=3.1.2
```
Comment (bdice) [inline recipes/scanpy/meta.yaml:30]: I think it’s recommended to depend on `matplotlib-base` if you don’t require the Qt frontend and only need the core parts of the library. See https://github.com/conda-forge/conda-forge.github.io/issues/933

Comment (flying-sheep) [reply recipes/scanpy/meta.yaml]: OK, thank you!

**Comment by flying-sheep** [comment]
That test failure isn’t mine. Who can restart that build?

**Comment by flying-sheep** [comment]
Ha, ready!

**Comment by flying-sheep** [comment]
Hi @conda-forge/staged-recipes, this is ready, could I please have some eyes? :smiley:

File: recipes/scanpy/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+test:
+  imports:
+    - scanpy
+  commands:
+    - scanpy
+  requires:
+    - pip
+  # commands:
+  #   - pip check  # TODO: fails because of xlrd not being installed, needs to be fixed for anndata
```
Comment (chrisburr) [inline recipes/scanpy/meta.yaml:59]: @flying-sheep Can you fix this and add a `pip check` test to `anndata`?

**Comment by chrisburr** [comment]
> That test failure isn’t mine. Who can restart that build?

You can! 😄 https://conda-forge.org/docs/maintainer/infrastructure.html#conda-forge-admin-please-restart-ci

**Pull request merged**